consensus.go

```diff
@@ -762,3 +762,21 @@ func exponentialTimeout(round uint64) time.Duration {
 	}
 	return timeout
 }
+
+// Calculate max faulty nodes in order to have Byzantine-fault tollerant system.
+// Formula explanation:
+// N -> number of nodes in PBFT
+// F -> number of faulty nodes
+// N = 3 * F + 1 => F = (N - 1) / 3
+
+// PBFT tolerates 1 failure with 4 nodes
+// 4 = 3 * 1 + 1
+// To tolerate 2 failures, PBFT requires 7 nodes
+// 7 = 3 * 2 + 1
+// It should always take the floor of the result
+func MaxFaultyNodes(nodesCount int) int {
```

Comment:
Can you add unit tests for this?

Reply:
There already was a unit test for this part, just added `TotalNodesCount=0` in order to cover that sanity check logical branch within the calculation as well.